Textfield/Textfield.jsx

```diff
@@ -3,49 +3,120 @@ import MaterialComponent from "../MaterialComponent";
 import { MDCTextfield } from "@material/textfield/";
 
 /**
+ * @prop persistent = false
+ * @prop validation-msg = false
+ */
+class Helptext extends MaterialComponent {
+  constructor() {
+    super();
+    this.componentName = "textfield-helptext";
+    this._mdcProps = ["persistent", "validation-msg"];
+  }
+  materialDom(props) {
+    return (
+      <p {...props} aria-hidden="true">
+        {props.children}
+      </p>
+    );
+  }
+}
+
+/**
  * @prop fullwidth = false
  * @prop multiline = false
+ * @prop dense = false
+ * @prop disabled = false
  * @prop type = 'text'
  */
-export default class Textfield extends MaterialComponent {
+class TextfieldInput extends MaterialComponent {
   constructor() {
     super();
     this.componentName = "textfield";
-    this.state = {
-      showFloatingLabel: false
-    };
-    this._mdcProps = ["fullwidth", "multiline"];
+    this._mdcProps = ["fullwidth", "multiline", "dense", "disabled"];
   }
   componentDidMount() {
-    this.setState({
```

Comment:
Right, I got that, but the issue is that the state in ```componentDidMount``` is changed after ```this.MDComponent``` is instanciated, which means that the label only appears after ```MDCTextfield``` has set the label to ```null``` because it didn't exist, which means that ```material-components-web``` doesn't update the label and add the floating class on focus.
I'm on the phone, not quite sure if what I'm writing makes sense, but I'm pretty sure that it didn't work before. Just check on the documentation, you'll see that the label never floats up.